build/git.go

```diff
@@ -111,5 +114,18 @@ func getGitAttributes(ctx context.Context, contextPath string, dockerfilePath st
 		}
 	}
 
-	return
+	return res, func(so *client.SolveOpt) {
+		if root == "" {
+			root = wd
+		}
+		for k, dir := range so.LocalDirs {
+			d, err := filepath.Abs(dir)
+			if err != nil || !strings.HasPrefix(d, root) {
+				continue
+			}
+			if r, err := filepath.Rel(root, d); err == nil {
+				so.FrontendAttrs["vcs-localdir:"+k] = r
```

Comment:
This needs to start with `vcs:`. There is already a rule in buildkit to put all such metadata to provenance.